app/components/blacklight/date_facet_hierarchy_component.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# Display the date facet in a tree hierarchy
+module Blacklight
+  class DateFacetHierarchyComponent < Blacklight::FacetFieldListComponent
+    def facet_item_presenters
+      facet_item_tree_hierarchy_from_year_list.map do |item|
+        facet_item_presenter(item)
+      end
+    end
+
+    private
+
+    # Build out the tree hierarchy (century -> decade -> year) for
+    # a facet and return the top-level facet items
+    #
+    # @return [Array<Blacklight::Solr::Response::Facets::FacetItem>]
+    def facet_item_tree_hierarchy_from_year_list
```

Comment:
I have read through this method and understand what you're doing but it is the trickiest part of the code. It might benefit from a comment about what format input is expected, and also about the different cases you handle along the way.

Reply:
good call, will do

Reply:
i may have gone too far in the other direction, but i added what should hopefully be a thorough explanation of expected input and output structures 😅

Reply:
Ty!!